website/docs/tutorials/07-testing/02-cheatcodes/reflect.md

```diff
@@ -0,0 +1,130 @@
+# `reflect`
+```python
+def reflect(self, ids: VmConsts) -> Reflector:
+
+class Reflector:
+    def get(self)
+```
+Loads specified Cairo object into Python type. ```get()``` can return:
+
+- int (for felt)
+- RelocatableValue (for pointer)
+- Cairo Struct (for more complex types)
+
+You can use it to print cairo data and compare complex structures.
+
+```cairo
+%lang starknet
+
+struct SimpleStruct:
+    member x: felt
+end
+
+@external
+func test_reflect_simple():
+    alloc_locals
+
+    local simple_struct: SimpleStruct = SimpleStruct(x=10)
+
+    %{
+        simple_struct = reflect(ids).simple_struct.get()
+        print(simple_struct)
+        # output:
+        # CairoStruct(
+        #     x=10
+        # )
+
+        assert simple_struct.x == 10 # true
+    %}
+
+    return()
+end
+```
+
+```cairo
+%lang starknet
+
+from starkware.cairo.common.registers import get_fp_and_pc
+
+@external
+func test_pointers():
+    alloc_locals
+
+    let (__fp__, _) = get_fp_and_pc()
+
+    local pointee: felt = 13
+    local ptr1: felt* = &pointee
+    local ptr2: felt* = &pointee
+    
+    %{
+        ptr1 = reflect(ids).ptr1.get()
+        ptr2 = reflect(ids).ptr2.get()
+
+        print(type(ptr1)) # RelocatableValue
+        assert ptr1 == ptr2  # Pointers are compared directly using their addresses
+    %}
+    return ()
+end
+```
+
+```cairo
+%lang starknet
+
+struct InnerStruct:
+    member value: felt
+end
+
+struct OuterStruct:
+    member inner_struct: InnerStruct
+end
+
+@external
+func test_nesting():
+    alloc_locals
+    local inner_struct: InnerStruct = InnerStruct(value=7)
+    local outer_struct: OuterStruct = OuterStruct(inner_struct=inner_struct)
+
+    %{
+        outer_struct = reflect(ids).outer_struct.get()
+        OuterStruct = CairoStruct #
+        InnerStruct = CairoStruct # This way you can add aliases for readability
+
+        # You can compare nested structs
+        assert outer_struct == OuterStruct(
+            inner_struct=InnerStruct(
+                value=7
+            )
+        )
+    %}
+    return ()
+end
+```
+
+```cairo
+%lang starknet
+
+struct TwoFieldStruct:
+    member value1: felt
+    member value2: felt
+end
+
+@external
+func test_wildcards():
+    alloc_locals
+    local two_field_struct: TwoFieldStruct = TwoFieldStruct(value1=23, value2=17)
+    
+    %{
+        two_field_struct = reflect(ids).two_field_struct.get()
+        assert two_field_struct == CairoStruct(
+            value1=23,
+            value2=two_field_struct.value2
+            # You can use struct members in comparison to make sure it evaluates to true
+        )
+    %}
+    return ()
+end
+```
+
+:::warning
+Unlike `ids`, `reflect` does not automatically dereference pointers. As of right now you have to dereference them in Cairo.
```

Comment:
> You have to dereference them in Cairo.

I think "as of right now" means something different

Reply:
```suggestion
Unlike `ids`, `reflect` does not automatically dereference pointers.
As of now, dereference them in Cairo first.
```